client/components/suggestions/item.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+class Item extends PureComponent {
+
+	static propTypes = {
+		label: PropTypes.string.isRequired,
+		hasHighlight: PropTypes.bool,
+		query: PropTypes.string,
+		onMouseDown: PropTypes.func.isRequired,
+		onMouseOver: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		hasHighlight: false,
+		searchTerm: '',
+	};
+
+	/**
+	 * Highlights the part of the text that matches the query.
+	 * @param  {string} text  Text.
+	 * @param  {string} query The text to be matched.
+	 * @return {element}      A React element including the highlighted text.
+	 */
+	createTextWithHighlight( text, query ) {
+		const re = new RegExp( '(' + query + ')', 'gi' );
+		const parts = text.split( re );
+
+		return parts.map( ( part, i ) => {
+			const key = text + i;
+			const lowercasePart = part.toLowerCase();
+			const spanClass = classNames( 'suggestions__label', {
+				'is-emphasized': lowercasePart === query,
+			} );
+
+			return <span key={ key } className={ spanClass } >{ part }</span>;
+		} );
+	}
+
+	handleMouseDown = ( event ) => {
+		event.stopPropagation();
+		event.preventDefault();
+
+		this.props.onMouseDown && this.props.onMouseDown( this.props.label );
```

Comment:
If `onMouseDown` is a required prop, then we should be able to call `this.props.onMouseDown` outright without checking that it's defined first.